pymeasure/instruments/keithley/keithley2281S.py

```diff
@@ -0,0 +1,573 @@
+        """Get measurement status."""
+        return Keithley2281SOperationEventRegister.MEASUREMENT in self.summary_event
+
+    @property
+    def reading_available(self) -> bool:
+        """Get availability of a reading."""
+        return Keithley2281SMeasurementEventRegister.READING_AVAILABLE in self.measurement_event
+
+    def characterize(
+        self,
+        lower_voltage: float,
+        upper_voltage: float,
+        charge_current: float,
+        memory_slot: int,
+        model_voltage_offset: float = 0.05,
+        charge_delay: float = 0,
+    ):
+        """
+        Convenience function for testing a battery and saving its model to the internal memory.
+
+        The device can only discharge the battery at a fixed 1A! If this current is too high,
+        a series resistor has to be used during discharge to limit the current!
+        If the battery is discharged below the lower limit, it will be charged with a 10th of
+        the set charge current till it reaches the lower limit, then the battery profile will
+        be characterized.
+
+        :param lower_voltage: discharge end voltage, set this slightly lower (~0.05V)
+                              than in normal operation
+        :type lower_voltage: float
+        :param upper_voltage: charge end voltage, set this slightly higher (~0.05V)
+                              than in normal operation
+        :type upper_voltage: float
+        :param charge_current: maximum charge current.
+                               A 1/100th is used as (dis-)charge end current.
+        :type charge_current: float
+        :param memory_slot: Internal memory slot to save the model to
+        :type memory_slot: int
+        :param model_voltage_offset: Voltage offset for the generated model
+                                     upper and lower limits. These have to be in
+                                     the range of the actual measured values, i.e.
+                                     in between the hard limits of the measurement.
+                                     Defaults to 0.05
+        :type model_voltage_offset: float, optional
+        :param charge_delay: Seconds to wait between discharging and charging.
+                             Actual time is higher since the device has some internal
+                             delay on reporting end of a test. Defaults to 0.
+        :type charge_delay: float, optional
+        """
+
+        # Reset device to known state
+        self.reset()
+        self.function_mode = "TEST"
+
+        # Set discharging conditions and start discharging
+        self.bt.charge_voltage_setpoint = lower_voltage
+        self.bt.charge_current_limit = charge_current / 10
+        self.bt.termination_current = charge_current / 100
+        self.bt.output_enabled = True
+        self.display_text_data = "Measurement Ongoing! Discharging..."
+        while self.measurement_ongoing:
+            time.sleep(1)
```

Comment:
I think it is good to mention in the docstring, that this method blocks until the whole measurement has finished.